eldritch/test_events.py

```diff
@@ -120,6 +148,123 @@ def testMoveMultipleSpaces(self):
     self.assertEqual(self.char.place.name, "Graveyard")
     self.assertEqual(self.char.movement_points, 1)
 
+  def testIllegalMoveMultipleSpaces(self):
+    self.char.movement_points = 1
+    movement = Movement(
+        self.char, [self.state.places[name] for name in ["Easttown", "Rivertown", "Graveyard"]])
+    self.assertFalse(movement.is_resolved())
+    self.assertEqual(self.char.movement_points, 1)
+
+    self.state.event_stack.append(movement)
+    self.resolve_until_done()
+
+    self.assertTrue(movement.is_resolved())
+    self.assertEqual(self.char.place.name, "Easttown")
+    self.assertEqual(self.char.movement_points, 0)
+
+  def testMoveMultipleThroughMonster(self):
+    self.char.fight_will_slider = 2
+    cultist = next(monster for monster in self.state.monsters if monster.name == "Cultist")
+    cultist.place = self.state.places["Easttown"]
+    movement = Movement(
+        self.char, [self.state.places[name] for name in ["Easttown", "Rivertown", "Graveyard"]])
+    self.assertFalse(movement.is_resolved())
+    self.assertEqual(self.char.movement_points, 4)
+
+    self.state.event_stack.append(Sequence([movement, EndMovement(self.char)], self.char))
+    choice = self.resolve_to_choice(MultipleChoice)
+    self.assertEqual(choice.choices, ["Fight", "Evade"])
+    choice.resolve(self.state, "Fight")
+
+    next_choice = self.resolve_to_choice(MultipleChoice)
+    self.assertEqual(next_choice.choices, ["Fight", "Flee"])
+    next_choice.resolve(self.state, "Fight")
+
+    third_choice = self.resolve_to_choice(CombatChoice)
+    self.assertIsNone(third_choice.choices)
+    third_choice.resolve(self.state, [])
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      self.resolve_until_done()
+
+    self.assertTrue(movement.is_resolved())
+    self.assertEqual(self.char.place.name, "Easttown")
+    # self.assertEqual(self.char.movement_points, 0)
+    # TODO: ^ should this pass?
```

Comment:
Okay, I figured this out. When you run the unit tests, it actually starts with turn_phase == "upkeep". That means that when you use EndMovement (a subclass of EndTurn), it actually advances from upkeep to movement. When it advances from upkeep to movement, it resets the character's movement points.

I recommend starting these tests in the movement phase.